### PR TITLE
Flysky Hall Stick MOD using UART4 by disabling stick A/D, resolving #140.

### PR DESCRIPTION
**Pull request by richardclli**

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -1102,7 +1104,7 @@ void getADC()
 
 #if defined(FLYSKY_HALL_STICKS)
     if (x < 4)
-      v = get_hall_adc_value(x) >> (1 - ANALOG_SCALE);
+      v = get_flysky_hall_adc_value(x) >> (1 - ANALOG_SCALE);
```
Comment (raphaelcoeffic) [inline radio/src/opentx.cpp:1107]: Any way you could hide this behind the normal function?

File: radio/src/targets/horus/hal.h [hunk 2]
```diff
@@ -282,10 +282,17 @@
...
-  #define ADC_GPIO_PIN_STICK_LV         GPIO_Pin_1      // PA.01
-  #define ADC_GPIO_PIN_STICK_RH         GPIO_Pin_2      // PA.02
-  #define ADC_GPIO_PIN_STICK_RV         GPIO_Pin_3      // PA.03
+  #if !defined(FLYSKY_HALL_STICKS)
+    #define ADC_GPIO_PIN_STICK_LH         GPIO_Pin_0      // PA.00
+    #define ADC_GPIO_PIN_STICK_LV         GPIO_Pin_1      // PA.01
+    #define ADC_GPIO_PIN_STICK_RH         GPIO_Pin_2      // PA.02
+    #define ADC_GPIO_PIN_STICK_RV         GPIO_Pin_3      // PA.03
+  #else
+    #define ADC_GPIO_PIN_STICK_LH           0
```
Comment (raphaelcoeffic) [inline radio/src/targets/horus/hal.h:291]: If possible, please try to completely undef these.

**Review by raphaelcoeffic — CHANGES_REQUESTED** [review]

**Comment by raphaelcoeffic** [comment]
@richardclli thx for targets! Are you still on the 2 other items? Any problems with them?

File: radio/src/opentx.cpp (hunk 1 above)
Comment (richardclli) [reply radio/src/opentx.cpp:1107]: I am not sure where should I hide, this code is currently comes from NV14, I just try to refactorize the driver to horus, I think NV14 firmware should work on this change as well and finally we can have NV14 target removed.

File: radio/src/targets/horus/hal.h (hunk 2 above)
Comment (richardclli) [reply radio/src/targets/horus/hal.h:291]: I will see if I can work out something more clean.

Comment (raphaelcoeffic) [reply radio/src/targets/horus/hal.h:291]: Just to say, many places rely on patterns like `#if defined(SOME_PIN_PORT)`, which would be defeated by such a definition set to `0` (still defined).

File: radio/src/opentx.cpp (hunk 1 above)
Comment (raphaelcoeffic) [reply radio/src/opentx.cpp:1107]: Oh! I see now! Sorry for not checking better earlier! Ok let's keep it like that, we can make it better later.

File: radio/src/targets/horus/hal.h (hunk 2 above)
Comment (richardclli) [reply radio/src/targets/horus/hal.h]: I have changed the way to disable ADC, please check if it is ok now.

Comment (raphaelcoeffic) [reply radio/src/targets/horus/hal.h]: Thx! we're good to go!

**Review by raphaelcoeffic — APPROVED** [review]

**Comment by richardclli** [comment]
So this will be merged?  Nice.

**Pull request merged**